pkg/webhook/merge_on_review_approval_and_status_checks.go

```diff
@@ -40,27 +40,41 @@ func (h *autoMerger) HandleEvent(w http.ResponseWriter, payload interface{}, ghC
 		return h.handlePullRequestEvent(w, event, ghClientFunc)
 	case *github.StatusEvent:
 		return h.handleStatusEvent(w, event, ghClientFunc)
+	case *github.PullRequestReviewEvent:
+		return h.handlePullRequestReviewEvent(w, event, ghClientFunc)
 	default:
 		return nil
 	}
 }
 
+func (h *autoMerger) handlePullRequestReviewEvent(w http.ResponseWriter, event *github.PullRequestReviewEvent, ghClientFunc GitHubIntegrationsClientFunc) error {
+	if strings.ToLower(event.Review.GetState()) != approvedReviewState {
+		return nil
+	}
+
+	return h.mergePRFromPullRequestEvent(event.Installation.GetID(), event.Repo, event.PullRequest, ghClientFunc)
+}
+
 func (h *autoMerger) handlePullRequestEvent(w http.ResponseWriter, event *github.PullRequestEvent, ghClientFunc GitHubIntegrationsClientFunc) error {
 	if strings.ToLower(event.GetAction()) != labeledEvent {
```

Comment:
The event doesn't contain what label has been added and you have to get the issue (a PR is also an issue...) which contains the list of labels... this is done in the called merge function so no need to repeat that here.